vnext/Desktop/BeastWebSocketResource.cpp

```diff
@@ -148,89 +141,102 @@ void BaseWebSocketResource<Protocol, SocketLayer, Stream, Resolver>::PerformWrit
 
   // Auto-fragment disabled. Adjust write buffer to the largest message length
   // processed.
-  if (request.first.length() > m_stream->write_buffer_size())
-    m_stream->write_buffer_size(request.first.length());
+  if (request.first.length() > m_stream->write_buffer_bytes())
+    m_stream->write_buffer_bytes(request.first.length());
 
-  m_stream->async_write(buffer(request.first), [this](boostecr ec, size_t size) {
-    if (ec) {
-      if (m_errorHandler)
-        m_errorHandler({ec.message(), ErrorType::Send});
-    } else {
-      if (m_writeHandler)
-        m_writeHandler(size);
-    }
+  m_stream->async_write(
+      buffer(request.first), bind_front_handler(&BaseWebSocketResource<SocketLayer, Stream>::OnWrite, SharedFromThis()));
+}
 
-    m_writeInProgress = false;
+template <typename SocketLayer, typename Stream>
+void BaseWebSocketResource<SocketLayer, Stream>::OnWrite(error_code ec, size_t size) {
```

Comment:
OOC do the changes here accrue to the WinRT stack or are they just for the legacy implementation?

Reply:
Totally unrelated to the WinRT implementation, but the Boost upgrade brought breaking API changes (and tons of benefits).